components/tabs/src/sync/store.rs

```diff
@@ -137,19 +140,14 @@ impl<'a> Store for TabsStore<'a> {
             let (client_name, device_type) = self
                 .remote_clients
                 .borrow()
-                .get(local_id)
+                .get(&local_id)
                 .map(|client| {
                     (
                         client.device_name.clone(),
                         client.device_type.unwrap_or(DeviceType::Mobile),
                     )
                 })
-                .unwrap_or_else(|| {
-                    // TODO: Since we know our own client ID thanks to
-                    // `self.remote_clients`, we don't need consumers to
-                    // pass it themselves.
-                    (String::new(), DeviceType::Mobile)
-                });
+                .unwrap_or_else(|| (String::new(), DeviceType::Mobile));
```

Comment:
I wonder, do you think we shouldn't upload tabs at all if we don't have info about the client? (Say, if the caller is using a legacy API and passed a garbage `local_device_id` to `sync`?) Or is it better to still upload, and just show a blank device name on the others?

Reply:
I'd rather show something in this case yes! But we can revisit the decision later though :)